lib/strings.nix

```diff
@@ -678,4 +678,36 @@ rec {
        => "1.0"
   */
   fileContents = file: removeSuffix "\n" (builtins.readFile file);
+
+
+  /* Creates a valid derivation name from a potentially invalid one.
+
+     Type: sanitizeDerivationName :: String -> String
+
+     Example:
+       sanitizeDerivationName "../hello.bar # foo"
+       => "-hello.bar-foo"
+       sanitizeDerivationName ""
+       => "unknown"
+       sanitizeDerivationName pkgs.hello
+       => "-nix-store-2g75chlbpxlrqn15zlby2dfh8hr9qwbk-hello-2.10"
+  */
+  sanitizeDerivationName = lib.flip lib.pipe [
```

Comment:
How much overhead do these compose function actually add?

Reply:
Shouldn't be too much. Though I have a change to it here which would improve performance a bit: https://github.com/NixOS/nixpkgs/pull/78640/commits/8022565db4a7f983ad9918990f9bd2e4a2a4a453